src/frame/mod.rs

```diff
@@ -239,6 +250,22 @@ impl From<Result<Response, ExceptionResponse>> for ResponsePdu {
     }
 }
 
+impl<E> TryFrom<Option<E>> for ResponsePdu
+where
+    E: Into<ResponsePdu>,
+{
+    type Error = std::io::Error;
+    fn try_from(from: Option<E>) -> Result<ResponsePdu, Self::Error> {
+        match from {
+            Some(e) => Ok(e.into()),
+            None => Err(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "empty response",
```

Comment:
empty response? `None` = no response.

This `TryFrom<Option<..>>` implementation might be convenient, but it seems confusing. Is it really needed?

Where is the context that justifies to reinterpret a neutral `None` as an arbitrary `io::Error`. The generic `TryFrom` trait implementation should not require such context outside of `ResponsePdu`.

Reply:
ok, I'll change the api to use `Option<Into<ResponsePdu>>` instead of the tryfrom then